From a92a20a4c5fb89ae5d001b454e652ce4d49a8466 Mon Sep 17 00:00:00 2001
From: AdarshGrewal <adarshgrewal@gmail.com>
Date: Sun, 24 Nov 2024 18:52:58 +0530
Subject: [PATCH 3/3] Camera: Avoid roundBufferDimensionsNearest also for
 HAL_PIXEL_FORMAT_Y16

 * used by miuicamera in portrait mode on recent device eg zircon

Change-Id: I78b1f044356aa61411ab7ce8a25c3c4e53c56afa
---
 .../camera/libcameraservice/utils/SessionConfigurationUtils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/camera/libcameraservice/utils/SessionConfigurationUtils.cpp b/services/camera/libcameraservice/utils/SessionConfigurationUtils.cpp
index bd9b5f40c8..c7a98ecdba 100644
--- a/services/camera/libcameraservice/utils/SessionConfigurationUtils.cpp
+++ b/services/camera/libcameraservice/utils/SessionConfigurationUtils.cpp
@@ -227,7 +227,7 @@ bool roundBufferDimensionNearest(int32_t width, int32_t height,
     // requirement. AIDE2 is vendor enhanced feature which requires special resolutions and
     // those are not populated in static capabilities.
     if (isPriviledgedClient == true &&
-            (format == HAL_PIXEL_FORMAT_YCbCr_420_888 || format == HAL_PIXEL_FORMAT_BLOB)) {
+            (format == HAL_PIXEL_FORMAT_YCbCr_420_888 || format == HAL_PIXEL_FORMAT_BLOB || format == HAL_PIXEL_FORMAT_Y16)) {
         ALOGI("Bypass roundBufferDimensionNearest for privilegedClient YUV streams "
                 "width %d height %d for format %d", width, height, format);
 
-- 
2.47.1

